src/app/header/header.component.ts

```diff
@@ -11,8 +12,11 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
+
 export class HeaderComponent {
-  constructor(private router: Router) {}
+
+  constructor(private router: Router,
+    private queryData: QueryService) { }
```

Comment:
queryService would be a better name for this variable.

Reply:
changed it to queryService

Reply:
Now the instance is called different of the class. Im not a fan of that